### ISSUE TITLE
World Freezes/ hangs forever when in/near ocean structures or exploring oceans for long periods of time 

### ISSUE THREAD

**Issue by The-Bernician-Lamb** [Serilum/.issue-tracker#100]
## **Information**
Minecraft version: 1.15.2
Forge version: 31.2.0
Environment: Singleplayer and Multiplayer

Mod name: Zombie Horse Spawn
Mod version: 2.1.2

## **Description**
I originally discovered this bug on a modded server I was testing and thought it was related to a mod called Strange as it always happened whenever I went near some of it's underground ocean structures, but I found that this issue occurs even when no other mods are installed. This always appears to happen when I am exploring oceans and the structures that they contain (Ocean ruins, Monuments, Shipwrecks). 

To replicate this issue, I use /locate to find Ocean ruins or monuments, after teleporting to them the game loads a few chunks (or sometimes none at all) and then stops loading them, all entities freeze (though I am still able to move around freely within the loaded chunks) and the server kicks me out after a minute or two. (In singleplayer, the same thing happens just without kicking me out). After this occurs the server does not respond to any input and I have to use task manager to force it to close. 

This always seems to happen after I find a Ocean ruin or a location with a lot of drowned (where they spawn as a part of the structure), This issue usually seems to occur fairly quickly in more recent tests but it sometimes takes a couple of structures before the issue is triggered and I have to force my game to close.

I originally reported this issue to the author of Strange, said issue can be found here if you want to read the discussion I had about it with the author: https://github.com/svenhjol/Strange/issues/40

## **Crash report**
N/A  (Game doesn't crash, just freezes/hangs forever)

**Comment by The-Bernician-Lamb** [comment]
To give a bit more context to this issue and what happens when it occurs I took a screenshot.

When the issue occurs, an abrupt chunk boundary appears like this:
![2020-05-17_23 18 22](https://user-images.githubusercontent.com/64381954/82161477-f61e8b00-9894-11ea-8b2f-2bbc9b066eb4.png)

Beyond this boundary, chunks refuse to load and no other chunks behind the boundary load either, I am able to move around freely, even into the unloaded chunks but this does not cause any more chunks to be loaded. 

In singleplayer, my world becomes stuck like this forever and I then have to close the game and delete the world as a result.

In multiplayer, the server/world becomes stuck like this, then kicks everyone out after a minute or two, giving the message "[player] Timed out", giving the same message when trying to reconnect. After this point the server stops responding to any input and I have to forcibly close it and delete the world and create a new one.

I hope this helps you understand this issue better.

**Comment by svenhjol** [comment]
Hi there, author of Strange here.

This ticket (https://github.com/svenhjol/Strange/issues/40) is the issue in question.

I am able to repeat this issue locally. There isn't any debug information to help me start working on it.

Ruins in my mod are using the vanilla jigsaw structure method and no mobs spawn by default - there are proximity sensor blocks that spawn them in when the player is inside.
@ricksouth would you be agreeable to provide source code? I haven't done any deeper investigation yet but it would be good to get compatiblity working :)

**Comment by jamiewood678** [comment]
I can also confirm this problem. Thank you @NJL5390 for identifying this, I couldn't figure out why my modpack wasn't working until you found this.

**Comment by The-Bernician-Lamb** [comment]
So, An update regarding this issue is that after I did some testing with the latest version of Zombie Horse Spawn (as well as Collective) and the latest forge version, I don't seem to be able to replicate this issue anymore. I will need to do more testing to definitely verify that this is the case so I will give an update on that soon but early signs look promising.

**Comment by The-Bernician-Lamb** [comment]
After some further testing on both singleplayer and multiplayer I have found that this issue seems to have disappeared altogether and no longer occurs. No matter how many chunks I generate and ocean structures that I visit, everything works fine and doesn't freeze anymore.

Therefore I have decided to close this issue. If I ever come across this issue (or a similar one) again in the future I may re-open this issue but for now consider it fixed. 

Thank you to @svenhjol and @ricksouth for putting the time in to fix this issue! You both make fantastic mods and I greatly appreciate the time and effort you both put into your mods!
